svglib/fonts.py

```diff
@@ -0,0 +1,199 @@
+        
+    def register_font_family(self, family, normal,  bold=None, italic=None, bolditalic=None):
+        self.register_font(family, normal)
+        if bold is not None:
+            self.register_font(family, bold, weight='bold')
+        if italic is not None:
+            self.register_font(family, italic, style='italic')
+        if bolditalic is not None:
+            self.register_font(family, bolditalic, weight='bold', style='italic')
+
+    def register_font(self, font_family, font_path=None, weight='normal', style='normal', rlgFontName=None):
+        """
+        Register a font identified by its family, weight and style linked to an actual fontfile. 
+        Or map an svg font family , weight and style combination to a reportlab fontname.
+        """
+        NOT_FOUND = (None, False)
+        internal_name = FontMap.build_internal_name(font_family, weight, style)
+        if rlgFontName is None: # if no reportlabs font name is given, use the internal fontname to register the reportlab font
+            rlgFontName = internal_name
+
+        if rlgFontName in STANDARD_FONT_NAMES: # mapping to one of the standard fonts, no need to register
+            self._map[internal_name] = {'svg_family': font_family, 'svg_weight': weight, 'svg_style': style, 'rlgFont': rlgFontName}
+            return internal_name, True
+        
+        if internal_name not in STANDARD_FONT_NAMES and font_path is not None:
+            try:
+                registerFont(TTFont(rlgFontName, font_path))
+                self._map[internal_name] = {'svg_family': font_family, 'svg_weight': weight, 'svg_style': style, 'rlgFont': rlgFontName}
+                return internal_name, True
+            except TTFError:
+                return NOT_FOUND
+        
+    def find_font(self, font_name, weight='normal', style='normal'):
+        """Return the font and a Boolean indicating if the match is exact."""
+        internal_name = FontMap.build_internal_name(font_name, weight, style)
+        # Step 1 check if the font is one of the buildin standard fonts
+        if internal_name in STANDARD_FONT_NAMES:
+            return internal_name, True
+        # Step 2 Check if font is already registered
+        elif internal_name in self._map.keys():
+            return self._map[internal_name]['rlgFont'], True
+        # Step 3 Try to auto register the font
+        else:
+            # Try first to register the font if it exists as ttf
+            guessed_filename = FontMap.guess_font_filename(font_name, weight, style)
+            reg_name, exact = self.register_font(font_name, guessed_filename)
+            if reg_name is not None:
+                return reg_name, exact
+            else:
+                return FontMap.use_fontconfig(font_name, weight, style)
+
+
+_font_map = FontMap() # the global font map
+
+def register_font(font_name, font_path=None, weight='normal', style='normal', rlgFontName=None):
```

Comment:
I thought about 2 usecases.
First one: use one font map for all conversions. I think this is the default use. In this case they get stored inside the global font_map and those functions you mentioned can be used.
Second one: The user wants a specific font map for one conversion or a set of conversions. This can be achieved by creating a FontMap instance and then passing it to SvgRenderer object as constructor parameter.